packages/neo-one-server-client/src/npmCheck.ts

```diff
@@ -0,0 +1,90 @@
+import { getEscapedNPMName } from '@neo-one/utils';
+import fetch from 'cross-fetch';
+import * as fs from 'fs-extra';
+
+export interface NPMResult {
+  readonly success: boolean;
+  readonly version: string;
+  readonly date: number;
+}
+const emptyNPMResult: NPMResult = {
+  success: false,
+  version: '',
+  date: 0,
+};
+
+const askNPM = async (pkg: string): Promise<NPMResult> =>
+  new Promise<NPMResult>(async (resolve) => {
+    const NPM_REGISTRY_URL = 'https://registry.yarnpkg.com';
+    const escapedName = getEscapedNPMName(pkg);
+    const url = `${NPM_REGISTRY_URL}/${escapedName}`;
+
+    await fetch(url).then(async (res) => {
+      if (res.ok) {
+        try {
+          const json = await res.json();
+          const version = json['dist-tags'].latest;
+          const npmResult: NPMResult = {
+            success: true,
+            version,
+            date: Date.parse(json.time[version]),
+          };
+          resolve(npmResult);
+        } catch {
+          // ignore errors...
+        }
+      }
+      resolve(emptyNPMResult);
+    });
+  });
+
+const promiseTimeout = async <t>(delay: number): Promise<t> =>
+  new Promise<t>((resolve) => {
+    setTimeout(resolve, delay, {});
```

Comment:
nit: we don't usually use this form, rather just use a lambda - `setTimeout(() => resolve({}), delay)`. Also, I don't think you want a generic here, just use `NPMResult`